src/providers/VoiceFocusProvider/index.tsx

```diff
@@ -0,0 +1,133 @@
+// Copyright 2020-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { useState, useEffect, useMemo, useContext, createContext, useCallback } from 'react';
+import { 
+    Device,
+    VoiceFocusTransformDevice,
+    VoiceFocusDeviceTransformer, 
+    VoiceFocusSpec, 
+    VoiceFocusDeviceOptions
+} from 'amazon-chime-sdk-js';
+
+interface Props {
+    /** Determines how how you want Amazon Voice Focus to behave */
+    spec?: VoiceFocusSpec;
+    /** Configuration for additional named arguments, including `logger` and `preload` */
+    options?: VoiceFocusDeviceOptions;
+}
+
+interface VoiceFocusState {
+    supportsVoiceFocus: boolean;
+    transformToVoiceFocus: (device: Device) => Promise<Device | VoiceFocusTransformDevice>;
+}
+
+const VoiceFocusContext = createContext<VoiceFocusState | null>(null);
+
+const VoiceFocusProvider: React.FC<Props> = ({
+    spec,
+    options,
+    children
+}) => {
+    const [supportsVoiceFocus, setSupportsVoiceFocus] = useState(false);
```

Comment:
Is there a pattern for delayed checks like this — start with `undefined` and then it changes to `true` or `false`?

Reply:
Make sense, set the default value as `undefined`.
There is always a warning on the console, `Amazon Voice Focus is not supported.` since `isVoiceFocusSupported` is false at the beginning. After changed it to `undefined`, the warning disappeared.